compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -276,7 +277,8 @@ pub(crate) struct CrateRoot {
 
     exported_symbols: LazyArray<(ExportedSymbol<'static>, SymbolExportInfo)>,
 
-    syntax_contexts: SyntaxContextTable,
+    syntax_context_keys: SyntaxContextKeyTable,
+    syntax_context_data: SyntaxContextDataTable,
```

Comment:
I don't understand why this field exists if the goal is to never encode/decode the auxiliary data in syntax contexts.
I expected to have only the `SyntaxContextKeyTable` (or equivalents) in both regular metadata and in incremental cache.

Reply:
It actually participates in the decode process (see [`decode_ctxt_data(d, raw_id)`](https://github.com/rust-lang/rust/pull/129827/files#diff-1f9c21b64fad30a12ac049ab77bfd50ab32a8b3e666bef801ebb906698bd1cfcR1433)). 

I think I need to find a way to compute the `SyntaxContextData` by `SyntaxContextKey` if we remove it.

Reply:
`fn apply_mark_internal` computes `SyntaxContextData` from `SyntaxContextKey` (and with a cache as well).